### PR TITLE
Expose data metrics per bucket space on content node

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review. Right now I'm just using a metric dimension key of "name" for the bucket space metrics, but might want to choose something else if these metrics are to be exported via the proxy.
@hakonhall FYI, this is intended to be used as input to the orchestrator logic when deciding if a retired node can be removed
@yngveaasheim FYI

Example dump of metrics with this change:
```
vds.datastored.alldisks.buckets average=33 last=33 min=33 max=33 count=2 total=66
vds.datastored.alldisks.docs average=34 last=34 min=34 max=34 count=2 total=68
vds.datastored.alldisks.bytes average=1984 last=1984 min=1984 max=1984 count=2 total=3968
vds.datastored.alldisks.activebuckets average=13 last=13 min=13 max=13 count=2 total=26
vds.datastored.alldisks.readybuckets average=18 last=18 min=18 max=18 count=2 total=36
vds.datastored.bucket_space{name:global}.buckets_total average=9 last=9 min=9 max=9 count=2 total=18
vds.datastored.bucket_space{name:global}.docs average=8 last=8 min=8 max=8 count=2 total=16
vds.datastored.bucket_space{name:global}.bytes average=432 last=432 min=432 max=432 count=2 total=864
vds.datastored.bucket_space{name:global}.active_buckets average=4 last=4 min=4 max=4 count=2 total=8
vds.datastored.bucket_space{name:global}.ready_buckets average=9 last=9 min=9 max=9 count=2 total=18
vds.datastored.bucket_space{name:default}.buckets_total average=24 last=24 min=24 max=24 count=2 total=48
vds.datastored.bucket_space{name:default}.docs average=26 last=26 min=26 max=26 count=2 total=52
vds.datastored.bucket_space{name:default}.bytes average=1552 last=1552 min=1552 max=1552 count=2 total=3104
vds.datastored.bucket_space{name:default}.active_buckets average=9 last=9 min=9 max=9 count=2 total=18
vds.datastored.bucket_space{name:default}.ready_buckets average=9 last=9 min=9 max=9 count=2 total=18
```
As can be observed above, legacy metrics that cover all bucket spaces remain unchanged.

### CODE REVIEW

**Comment by hakonhall** [comment]
The metric to use to decide whether a node can be removed (all documents/buckets moved away from the node) is vds.datastored.bucket_space{name:default}.buckets_total or sum_{x != global} vds.datastored.bucket_space{name:x}.buckets_total?

File: storage/src/vespa/storage/bucketdb/bucketmanagermetrics.cpp [hunk 1]
```diff
@@ -23,11 +25,28 @@ DataStoredMetrics::DataStoredMetrics(const std::string& name, metrics::MetricSet
     ready.logOnlyIfSet();
 }
 
-DataStoredMetrics::~DataStoredMetrics() { }
+DataStoredMetrics::~DataStoredMetrics() = default;
 
-BucketManagerMetrics::BucketManagerMetrics()
+BucketSpaceMetrics::BucketSpaceMetrics(const vespalib::string& space_name, metrics::MetricSet* owner)
+        : metrics::MetricSet("bucket_space", {{"name", space_name}}, "", owner),
```
Comment (geirst) [inline storage/src/vespa/storage/bucketdb/bucketmanagermetrics.cpp]: I think we should give dimension a more specific name. E.g 'bucketSpace'. Yes, we use camel case for dimension names.

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good, but I think we should rename dimension sooner rather than later.

File: storage/src/vespa/storage/bucketdb/bucketmanagermetrics.cpp (hunk 1 above)
Comment (vekterli) [reply storage/src/vespa/storage/bucketdb/bucketmanagermetrics.cpp]: Renamed `name` to `bucketSpace`

**Comment by vekterli** [comment]
@geirst PTAL
@hakonhall the idea was to use `vds.datastored.bucket_space{name:default}.buckets_total == 0` as a trigger for when a retired node can be removed, as that implies all non-global documents have been migrated away. This does leave an edge case where _global_ documents may be out of sync across replicas (and therefore theoretically may cause data loss if the retired node is removed), but we can add logic to the cluster controller for querying this if required. It has a holistic view of each bucket space's sync status in the cluster

**Review by geirst — APPROVED** [review]
👍

**Comment by hakonhall** [comment]
That's fine - the question was whether to sum over `x != global` or just pick out `x = default`. This has successfully rolled out and can now be used in orchestration?

**Pull request merged**